books/books/spiders/followall.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import re
+from six.moves.urllib.parse import urlparse
+import datetime
+import scrapy
+from scrapy.http import Request, HtmlResponse
+from scrapy.linkextractors import LinkExtractor
+import click
+from books.items import Page
+t = datetime.datetime.utcnow()
+
+
+class FollowAllSpider(scrapy.Spider):
+
+    name = 'followall'
+    ratings_map = {
+        'one': 1,
+        'two': 2,
+        'three': 3,
+        'four': 4,
+        'five': 5,
+    }
+
+    def __init__(self, **kw):
+        super(FollowAllSpider, self).__init__(**kw)
+        url = 'http://localhost/books.toscrape.com/index.html'
+        if not url.startswith('http://') and not url.startswith('https://'):
+            url = 'http://%s/' % url
+        self.url = url
+        self.allowed_domains = [re.sub(r'^www\.', '', urlparse(url).hostname)]
+        self.link_extractor = LinkExtractor()
+        self.cookies_seen = set()
+        self.previtem = 0
+        self.items = 0
+        self.timesec = datetime.datetime.utcnow()
+
+    def start_requests(self):
+        return [Request(self.url, callback=self.parse, dont_filter=True)]
+
+    def parse(self, response):
+        """Parse a PageItem and all requests to follow"""
+        page = self._get_item(response)
+        r = [page]
+
+        r.extend(self._extract_requests(response))
+        self.items = self.crawler.stats.get_value('item_scraped_count', 0)
+        pages = self.crawler.stats.get_value('response_received_count', 0)
+        a = self.crawler.stats.get_value('start_time')
+        b = datetime.datetime.utcnow()
+        self.timesec = b - a
+
+        if 280 <= self.items <= 300:
+            global t
+            t = datetime.datetime.utcnow()
+        if self.items > 300:
+            self.items = self.items - 300
+            self.timesec = b - t
+
+        return r
+
+    def close(self, reason):
+        f = open("AvSpeed.txt", 'a')
```

Comment:
I understand that you run the spider multiple times and need to append, but this seems a bit brittle - what if the file exists before the first run? It can happen if there is some error in run management code (``bench.py``), and the file is not removed. If you will do the loop in python instead of bash, then you can open the file in "w" mode, write the stats, close the file, then read them in bench.py and start another iteration.